servers/alzirr/overlays.nix

```diff
@@ -0,0 +1,17 @@
+[
+  (self: super:
+    let
+      ghc864 = self.callPackage (import
+        (builtins.fetchTarball
+          { url = "https://github.com/nixos/nixpkgs-channels/archive/nixos-19.03.tar.gz"; sha256 = "11z6ajj108fy2q5g8y4higlcaqncrbjm3dnv17pvif6avagw4mcb"; })
```

Comment:
Please make this a flake input

Reply:
I have no idea how I would go about setting the overlay in the flake and then passing it down. Would you mind?

Reply:
Just have this file be a function which takes `inputs`, and when importing this file, call it with `inputs`

Reply:
Oh, just remember we use this in `nixpkgs-overlays` and `environment.etc."nix/overlays.nix".source` too, it won't work with those, but I don't think we need those anymore. In case we do, we should have a better way regardless. Is there a way of making `nix repl` and `nix-shell`, etc use a package set we can define in the system flake somehow (as opposed to channel+overlay)?

Reply:
nix repl doesn't have flake support yet. `nix-shell` being an "oldschool" command, doesn't even know what a flake is either. So no, not really.

Reply:
Oh, actually, I think we could use a cool hack: export `legacyPackages` from this flake, add flake-compat, then do `nixpkgs=${inputs.self}` for this machine!

Reply:
Well, feel free to push :stuck_out_tongue:

Reply:
I like that idea and think it could come in handy another time too

Reply:
@mkaito @balsoft can either of you implement this?